src/app/helpers/helperMethods.ts

```diff
@@ -1,37 +1,146 @@
+const path = require('path');
+const hosts = [
+    "excel",
+    "onenote",
+    "outlook",
+    "powerpoint",
+    "project",
+    "word"
+];
 
 export namespace helperMethods {
-
-export function deleteFolderRecursively(projectFolder: string) 
-{
-    if(fs.existsSync(projectFolder))
+    export function deleteFolderRecursively(projectFolder: string) 
     {
-        fs.readdirSync(projectFolder).forEach(function(file,index){ 
-        var curPath = projectFolder + "/" + file; 
-        
-        if(fs.lstatSync(curPath).isDirectory())
+        if(fs.existsSync(projectFolder))
         {
-            deleteFolderRecursively(curPath);
+            fs.readdirSync(projectFolder).forEach(function(file,index){ 
+            var curPath = projectFolder + "/" + file; 
+            
+            if(fs.lstatSync(curPath).isDirectory())
+            {
+                deleteFolderRecursively(curPath);
+            }
+            else
+            {
+                fs.unlinkSync(curPath);
+            }
+        }); 
+        fs.rmdirSync(projectFolder); 
         }
-        else
+    }
+
+    export function doesProjectFolderExist(projectFolder: string)
+    {      
+    if (fs.existsSync(projectFolder))
         {
-            fs.unlinkSync(curPath);
+        if (fs.readdirSync(projectFolder).length > 0)
+        {          
+            return true;
         }
-    }); 
-    fs.rmdirSync(projectFolder); 
+        }
+        return false;
+    };
+
+    export function modifyProjectForSingleHost(projectFolder: string, projectType: string, host: string, typescript: boolean) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await convertProjectToSingleHost(projectFolder, projectType, host, typescript);
+                await updatePackageJsonForSingleHost(projectFolder, host);
+                return resolve();
+            } catch (err){
+                return reject(err);
+            }
+        });
     }
-};
 
-export function doesProjectFolderExists (projectFolder: string)
-{      
-  if (fs.existsSync(projectFolder))
-    {
-      if (fs.readdirSync(projectFolder).length > 0)
-      {          
-        return true;
-      }
+    async function convertProjectToSingleHost(projectFolder: string, projectType: string, host: string, typescript: boolean): Promise<void> {        
+        try {
+            // copy host-specific manifest over manifest.xml
+            const manifestContent: any = await readFileAsync(path.resolve(`${projectFolder}/manifest.${host}.xml`), 'utf8');
+            await writeFileAsync(path.resolve(`${projectFolder}/manifest.xml`), manifestContent);
+
+            switch (projectType) {
+                case "taskpane":
+                {
+                    // copy host-specific taskpane.ts[js] over src/taskpane/taskpane.ts[js]
+                    const srcContent = await readFileAsync(path.resolve(`${projectFolder}/src/taskpane/${host}.${typescript ? 'ts' : 'js'}`), 'utf8');
```

Comment:
You could use: `const extension = typescript ? "ts" : "js";` and then use extension in relevant places.

Reply:
good idea